sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -523,31 +523,14 @@ private Block BuildPatternExpression(PatternSyntaxWrapper patternSyntaxWrapper,
             }
             else if (RecursivePatternSyntaxWrapper.IsInstance(patternSyntaxWrapper))
             {
-                return BuildRecursivePatternExpression((RecursivePatternSyntaxWrapper)patternSyntaxWrapper, currentBlock);
+                // The recursive pattern will be handled in CSharpExplodedGraph and UcfgInstructionFactory.
+                currentBlock.ReversedInstructions.Add(patternSyntaxWrapper);
```

Comment:
Indeed. This is because we have this code:
```
                case SyntaxKindEx.RecursivePattern: // https://github.com/SonarSource/sonar-dotnet/issues/2937
                case SyntaxKindEx.ConstantPattern:
                    // The 0 in 'case 0 when ...'
                    // Do nothing
                    break;
```
in the `CSharpExplodedGraph` visit instructions method.